[FX] Assert None concrete_args and improve error messages

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #74736
* __->__ #74662

Previously, we would not emit a check that `concrete_args` with value `None` matched that value during runtime. This fixes that and improves some of the warning messages

Differential Revision: [D35137362](https://our.internmc.facebook.com/intern/diff/D35137362)

Commit:
[FX] Assert None concrete_args and improve error messages

[ghstack-poisoned]

Commit:
Update on "[FX] Assert None concrete_args and improve error messages"


Previously, we would not emit a check that `concrete_args` with value `None` matched that value during runtime. This fixes that and improves some of the warning messages


[ghstack-poisoned]

Commit:
Update on "[FX] Assert None concrete_args and improve error messages"


Previously, we would not emit a check that `concrete_args` with value `None` matched that value during runtime. This fixes that and improves some of the warning messages


[ghstack-poisoned]

jamesr66a commented:
@jamesr66a has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35137362).

Chillee reviewed (approved):
Makes sense for what it does.

I do kinda wonder whether these checks are a good idea in the first place though... I'm worried it just adds unnecessary complexity/don't always work.

Although, I'm guessing that you want these guards here since you find them helpful, in which case, ignore that :P

jamesr66a commented:
@Chillee yeah I was helping @pbelevich debug an issue in PiPPy that probably would have been more clear if we had these guards

Commit:
Update on "[FX] Assert None concrete_args and improve error messages"


Previously, we would not emit a check that `concrete_args` with value `None` matched that value during runtime. This fixes that and improves some of the warning messages

Differential Revision: [D35137362](https://our.internmc.facebook.com/intern/diff/D35137362)

[ghstack-poisoned]
